src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PXOverrideMismatch/PXOverrideMismatchAnalyzer.cs

```diff
@@ -0,0 +1,230 @@
+
+				var hasMatchingMethod = allTypes
+					.SelectMany(t => t.GetMembers().OfType<IMethodSymbol>())
+					.Any(m => IsCompatibleMethod(m, methodSymbol));
+
+				if (!hasMatchingMethod)
+				{
+					var diagnostic = Diagnostic.Create(
+						Descriptors.PX1096_PXOverrideMustMatchSignature,
+						methodSymbol.Locations.First());
+
+					context.ReportDiagnosticWithSuppressionCheck(diagnostic, pxContext.CodeAnalysisSettings);
+				}
+			}
+			else
+			{
+				// PXOverride can only be used in a PXGraphExtension child type.
+				var diagnostic = Diagnostic.Create(
+					Descriptors.PX1096_PXOverrideContainerIsNotPXGraphExtension,
+					methodSymbol.Locations.First());
+
+				context.ReportDiagnosticWithSuppressionCheck(diagnostic, pxContext.CodeAnalysisSettings);
+			}
+		}
+
+		private static HashSet<INamedTypeSymbol> GetAllBaseTypes(
+			List<INamedTypeSymbol> pxGraphExtensionTypes,
+			params INamedTypeSymbol[] types)
+		{
+			var allTypes = new HashSet<INamedTypeSymbol>(SymbolEqualityComparer.Instance);
+
+			foreach (var type in types)
+			{
+				GetBaseTypes(pxGraphExtensionTypes, type, allTypes);
+			}
+
+			return allTypes;
+		}
+		private static void GetBaseTypes(
+			List<INamedTypeSymbol> pxGraphExtensionTypes,
+			INamedTypeSymbol typeSymbol,
+			HashSet<INamedTypeSymbol> hashSet)
+		{
+			if (typeSymbol == null || typeSymbol.SpecialType != SpecialType.None || hashSet.Contains(typeSymbol))
+			{
+				return;
+			}
+
+			hashSet.Add(typeSymbol);
+
+			var extensionType = GetPXGraphExtension(pxGraphExtensionTypes, typeSymbol.BaseType);
+			var graphType = FindFirstTypeArgumentOfPxGraphExtension(extensionType);
+
+			if (graphType != null)
+			{
+				GetBaseTypes(pxGraphExtensionTypes, graphType, hashSet);
+			}
+
+			GetBaseTypes(pxGraphExtensionTypes, typeSymbol.BaseType, hashSet);
+		}
+		private static INamedTypeSymbol FindFirstTypeArgumentOfPxGraphExtension(INamedTypeSymbol type)
+		{
+			if (type == null)
+			{
+				return null;
+			}
+
+			if (type.TypeArguments.Any())
+			{
+				return type.TypeArguments.FirstOrDefault() as INamedTypeSymbol;
+			}
+
+			return FindFirstTypeArgumentOfPxGraphExtension(type.ContainingType);
+		}
+
+		private static INamedTypeSymbol GetPXGraphExtension(List<INamedTypeSymbol> pxGraphExtensionTypes, INamedTypeSymbol typeSymbol)
+		{
+			if (typeSymbol == null)
+			{
+				return null;
+			}
+
+			if (Implements(typeSymbol, pxGraphExtensionTypes) && typeSymbol.IsGenericType)
+			{
+				return typeSymbol;
+			}
+
+			return GetPXGraphExtension(pxGraphExtensionTypes, typeSymbol.BaseType);
+		}
+
+		private static bool Implements(INamedTypeSymbol typeSymbol, List<INamedTypeSymbol> baseTypes)
+		{
+			if (typeSymbol == null)
+			{
+				return false;
+			}
+
+			if (baseTypes.Any(b => typeSymbol.MetadataName == b.MetadataName))
+			{
+				return true;
+			}
+
+			return Implements(typeSymbol.BaseType, baseTypes);
+		}
+
+
+		private static bool IsCompatibleMethod(IMethodSymbol src, IMethodSymbol ext)
```

Comment:
IsCompatible method doesn't check if the method from the base type has declared accessibility `public`/`protected`, nor does it check if the method is `virtual`.  It also doesn't check method names.

Please use sensible names for parameters. And it could be good to refactor this method by splitting checks into several methods. I think you can reuse the parameters comparison which is done twice by making it a separate method.

Reply:
Thank you for your feedback. This method can be made cleaner, this is true. We can also add unit tests around it. I'll work on it.

One thing on the method name: I think Acumatica allows different method names to be px-overriden. We have examples on our code base.

Reply:
This might be a mistake from my side. I rerun the analyzer on our code base with the name constraint and it seems to be fine. I'll add the name check. Thank you.